src/msg/async/rdma/RDMAStack.cc

```diff
@@ -128,36 +128,60 @@ void RDMADispatcher::handle_async_event()
       return;
     }
     perf_logger->inc(l_msgr_rdma_total_async_events);
+    ldout(cct, 1) << __func__ << "Event : " << ibv_event_type_str(async_event.event_type) << dendl;
+
     switch (async_event.event_type) {
       /***********************CQ events********************/
       case IBV_EVENT_CQ_ERR:
-        lderr(cct) << __func__ << " CQ Overflow, dev = " << ib->get_device()->ctxt
+        lderr(cct) << __func__ << " Fatal Error, effect all QP bound with same CQ, "
+                   << " CQ Overflow, dev = " << ib->get_device()->ctxt
                    << " Need destroy and recreate resource " << dendl;
         break;
       /***********************QP events********************/
       case IBV_EVENT_QP_FATAL:
-        /* Error occurred on a QP and it transitioned to error state */
-        lderr(cct) << __func__ << " Error occurred on a QP and it transitioned to error state, dev = "
-                   << ib->get_device()->ctxt << " Need destroy and recreate resource " << dendl;
+        {
+          /* Error occurred on a QP and it transitioned to error state */
+          ibv_qp* ib_qp = async_event.element.qp;
+          uint32_t qpn = ib_qp->qp_num;
+          std::lock_guard l{lock};
```

Comment:
Why not to use get_qp() which implies lock?  QP won't disappear under us, because we are the thread which finally frees QP.

Reply:
Good catch.
Done. Wait for updating PR